File: scripting_languages/Ruby/Eulers/euler_tests.rb
#!/usr/bin/ruby

require_relative "Euler001"
require_relative "Euler002"
require_relative "Euler003"
require_relative "Euler004"
require_relative "Euler005"
require_relative "Euler006"
require_relative "Euler007"
require_relative "Euler008"
require_relative "Euler009"
require_relative "Euler010"
require "test/unit"
 
class TestEulers < Test::Unit::TestCase

    def test_euler_1_example
        euler = Euler001.new( 10 )
        assert_equal( 23, euler.run() )
    end
    
    def test_euler_1_actual
        euler = Euler001.new( 1000 )
        assert_equal( 233168, euler.run() )
    end

    def test_euler_2_example
        euler = Euler002.new( 100 )
        assert_equal( 44, euler.run() )
    end
    
    def test_euler_2_actual
        euler = Euler002.new( 4000000 )
        assert_equal( 4613732, euler.run() )
    end

    def test_euler_3_example
        euler = Euler003.new( 13195 )
        assert_equal( 29, euler.run() )
    end
    
    def test_euler_3_actual
        euler = Euler003.new( 600851475143 )
        assert_equal( 6857, euler.run() )
    end

    def test_euler_4_example
        euler = Euler004.new( 2 )
        assert_equal( 9009, euler.run() )
    end
    
    def test_euler_4_actual
        euler = Euler004.new( 3 )
        assert_equal( 906609, euler.run() )
    end

    def test_euler_5_example
        euler = Euler005.new( 10 )
        assert_equal( 2520, euler.run() )
    end
    
    def test_euler_5_actual
        euler = Euler005.new( 20 )
        assert_equal( 232792560, euler.run() )
    end
    
    def test_euler_6_example
        euler = Euler006.new( 10 )
        assert_equal( 2640, euler.run() )
    end
    
    def test_euler_6_actual
        euler = Euler006.new( 100 )
        assert_equal( 25164150, euler.run() )
    end
    
    def test_euler_7_example
        euler = Euler007.new( 6 )
        assert_equal( 13, euler.run() )
    end
    
    def test_euler_7_actual
        euler = Euler007.new( 10001 )
        assert_equal( 104743, euler.run() )
    end

    def test_euler_8_example
        euler = Euler008.new( 4 )
        assert_equal( 5832, euler.run() )
    end
    
    def test_euler_8_actual
        euler = Euler008.new( 13 )
        assert_equal( 23514624000, euler.run() )
    end
    
    def test_euler_9_example
        euler = Euler009.new( 12 )
        assert_equal( 60, euler.run() )
    end
    
    def test_euler_9_actual
        euler = Euler009.new( 1000 )
        assert_equal( 31875000, euler.run() )
    end

    def test_euler_10_example
        euler = Euler010.new( 10 )
        assert_equal( 17, euler.run() )
    end
    
    def test_euler_10_actual
        euler = Euler010.new( 2000000 )
        assert_equal( 142913828922, euler.run() )
    end
end

File: scripting_languages/Python_2_7/Eulers/euler_tests.py
#!/usr/bin/python

import unittest
from Euler001 import Euler001
from Euler002 import Euler002
from Euler003 import Euler003
from Euler004 import Euler004
from Euler005 import Euler005
from Euler006 import Euler006
from Euler007 import Euler007
from Euler008 import Euler008
from Euler009 import Euler009
from Euler010 import Euler010


class Euler001TestSuite(unittest.TestCase):
    def setUp(self):
        self.sut = Euler001()

    def test_sample(self):
        test_restrict = 10
        test_answer = 23
        self.assertEqual(self.sut(test_restrict), test_answer)

    def test_actual(self):
        test_restrict = 1000
        test_answer = 233168
        self.assertEqual(self.sut(test_restrict), test_answer)


class Euler002TestSuite(unittest.TestCase):
    def setUp(self):
        self.sut = Euler002()

    def test_sample(self):
        test_restrict = 10
        test_answer = 10
        self.assertEqual(self.sut(test_restrict), test_answer)

    def test_actual(self):
        test_restrict = 4000000
        test_answer = 4613732
        self.assertEqual(self.sut(test_restrict), test_answer)


class Euler003TestSuite(unittest.TestCase):
    def setUp(self):
        self.sut = Euler003()

    def test_sample(self):
        test_restrict = 13195
        test_answer = 29
        self.assertEqual(self.sut(test_restrict), test_answer)

    def test_actual(self):
        test_restrict = 600851475143
        test_answer = 6857
        self.assertEqual(self.sut(test_restrict), test_answer)


class Euler004TestSuite(unittest.TestCase):
    def setUp(self):
        self.sut = Euler004()

    def test_sample(self):
        test_restrict = 2
        test_answer = 9009
        self.assertEqual(self.sut(test_restrict), test_answer)

    def test_actual(self):
        test_restrict = 3
        test_answer = 906609


class Euler005TestSuite(unittest.TestCase):
    def setUp(self):
        self.sut = Euler005()

    def test_sample(self):
        test_restrict = 10
        test_answer = 2520
        self.assertEqual(self.sut(test_restrict), test_answer)

    def test_actual(self):
        test_restrict = 20
        test_answer = 232792560
        self.assertEqual(self.sut(test_restrict), test_answer)


class Euler006TestSuite(unittest.TestCase):
    def setUp(self):
        self.sut = Euler006()

    def test_sample(self):
        test_restrict = 10
        test_answer = 2640
        self.assertEqual(self.sut(test_restrict), test_answer)

    def test_actual(self):
        test_restrict = 100
        test_answer = 25164150
        self.assertEqual(self.sut(test_restrict), test_answer)


class Euler007TestSuite(unittest.TestCase):
    def setUp(self):
        self.sut = Euler007()

    def test_sample(self):
        test_restrict = 6
        test_answer = 13
        self.assertEqual(self.sut(test_restrict), test_answer)

    def test_actual(self):
        test_restrict = 10001
        test_answer = 104743
        self.assertEqual(self.sut(test_restrict), test_answer)


class Euler008TestSuite(unittest.TestCase):
    def setUp(self):
        self.sut = Euler008()

    def test_sample(self):
        test_restrict = 4
        test_answer = 5832
        self.assertEqual(self.sut(test_restrict), test_answer)

    def test_actual(self):
        test_restrict = 13
        test_answer = 23514624000
        self.assertEqual(self.sut(test_restrict), test_answer)


class Euler009TestSuite(unittest.TestCase):
    def setUp(self):
        self.sut = Euler009()

    def test_sample(self):
        test_restrict = 12
        test_answer = 60
        self.assertEqual(self.sut(test_restrict), test_answer)

    def test_actual(self):
        test_restrict = 1000
        test_answer = 31875000


class Euler010TestSuite(unittest.TestCase):
    def setUp(self):
        self.sut = Euler010()

    def test_sample(self):
        test_restrict = 10
        test_answer = 17
        self.assertEqual(self.sut(test_restrict), test_answer)

    def test_actual(self):
        test_restrict = 2000000
        test_answer = 142913828922
        self.assertEqual(self.sut(test_restrict), test_answer)


if __name__ == '__main__':
    unittest.main()
